From d85bdfbf075572d7f07b913b4bc63344110ecee5 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Mon, 27 Oct 2014 18:36:37 -0400
Subject: [x86] quirks: Print the Intel graphics stolen memory range

Message-id: <1414434997-7515-7-git-send-email-rclark@redhat.com>
Patchwork-id: 98479
O-Subject: [RHEL7 drm 6/6] x86/gpu: Print the Intel graphics stolen memory range
Bugzilla: 1154053
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

RHBZ: 1154053
Upstream: c71ef7b3c3be3337deaf1eb28dd26e0d5d4b4aa7

commit c71ef7b3c3be3337deaf1eb28dd26e0d5d4b4aa7
Author:     Ville Syrjala <ville.syrjala@linux.intel.com>
AuthorDate: Wed Feb 5 21:29:00 2014 +0200
Commit:     Ingo Molnar <mingo@kernel.org>
CommitDate: Sun Feb 9 15:11:31 2014 +0100

    x86/gpu: Print the Intel graphics stolen memory range

    Print an informative message when reserving the graphics stolen
    memory region in the early quirk.

    Signed-off-by: Ville Syrjala <ville.syrjala@linux.intel.com>
    Cc: Bjorn Helgaas <bhelgaas@google.com>
    Link: http://lkml.kernel.org/r/1391628540-23072-4-git-send-email-ville.syrjala@linux.intel.com
    Signed-off-by: Ingo Molnar <mingo@kernel.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/early-quirks.c b/arch/x86/kernel/early-quirks.c
index 5218dd2..52f36e6 100644
--- a/arch/x86/kernel/early-quirks.c
+++ b/arch/x86/kernel/early-quirks.c
@@ -520,6 +520,8 @@ static void __init intel_graphics_stolen(int num, int slot, int func)
    size = stolen_funcs->size(num, slot, func);
    start = stolen_funcs->base(num, slot, func, size);
    if (size && start) {
+    printk(KERN_INFO "Reserving Intel graphics stolen memory at 0x%x-0x%x\n",
+           start, start + (u32)size - 1);
     /* Mark this space as reserved */
     e820_add_region(start, size, E820_RESERVED);
     sanitize_e820_map(e820.map,
-- 
1.7.1